flight_path/flight_path.py

```
@@ -0,0 +1,105 @@
+import pandas as pd
+from geopy import Point
+from collections import defaultdict
+from geopy.distance import geodesic
+from geographiclib.geodesic import Geodesic
+# to visualize
+import folium
+'''
+Flight Path Component.
+
+This script calculates waypoints between two airports based on their ICAO/IATA codes.
+
+Features:
+    - Reads airport data from a CSV file. (This will need to be chnaged when merging with the driver!!!!)
+    - Retrieves coordinates for given airport codes. (Private)
+    - Computes equally spaced waypoints along a great-circle path. (We can choose how may points along the path we want)
+    - Uses GeographicLib for precise bearing calculations. (precise just means not in a straight line. Instead, this library takes into consideration the curvature of the earth!)
+
+'''
+
+class FlightPath:
+
+    airport_data =pd.read_csv("../airport_info/airports_data.csv")
+
+    def __init__(self, departure_code: str, destination_code: str):
+        self.departure = self.__get_coordinates(departure_code)
+        self.destination = self.__get_coordinates(destination_code)
+        pass
+    
+
+    """
+    Translates Airport Codes to Coordinates.
+
+    Args:
+        airport_code (str): Aiport Code (for now it is accepting both ICAC and IATA format)
+    
+    Returns:
+        (latitude, longitude) tuple(str): Tuple of latitude and longitude of the airport position
+    """
+    def __get_coordinates(self, airport_code: str):
+        row = self.airport_data.loc[(self.airport_data["ICAO"] == airport_code) | (self.airport_data["IATA"] == airport_code)] # This can be chnaged to only ICAC in the future when merging.
+
+        if row.empty:
+            print(f"Airport code {airport_code} not found!")
+            return None
```

I think we should raise an exception if we couldn't find the Airport location.

```__init__ ``` is not handling the None case right now but won't matter once  is this is changed to raise.